Tests/SwiftParserTest/translated/IfconfigExprTests.swift

```diff
@@ -500,26 +500,49 @@ final class IfconfigExprTests: XCTestCase {
     )
   }
 
-  // FIXME: Parsing should generate diagnostics - https://github.com/apple/swift-syntax/issues/1395
+  // FIXME: Diagnostics could be better.
   func testIfConfigExpr34() {
     assertParse(
       """
       #if MY_FLAG
-      #
+      #1️⃣
       elif
       #endif
-      """
+      """,
+      diagnostics: [
+        DiagnosticSpec(
+          message: "expected identifier in macro expansion",
+          fixIts: ["insert identifier"]
+        )
+      ],
+      fixedSource: """
+        #if MY_FLAG
+        #<#identifier#>
+        elif
+        #endif
+        """
     )
   }
 
-  // FIXME: Parsing should generate diagnostics - https://github.com/apple/swift-syntax/issues/1395
+  // FIXME: Diagnostics could be better.
   func testIfConfigExpr35() {
     assertParse(
       """
       #if MY_FLAG
-      # elif
+      # 1️⃣elif
       #endif
-      """
+      """,
+      diagnostics: [
+        DiagnosticSpec(
+          message: "expected identifier in macro expansion",
+          fixIts: ["insert identifier"]
+        )
+      ],
+      fixedSource: """
+        #if MY_FLAG
+        #<#identifier#>elif
```

Comment:
Lost the space here 🤔?

Reply:
Yeah.. no idea what happened..

Reply:
To start, this gets parsed as follows

```
SourceFileSyntax
IfConfigDeclSyntax
├─clauses: IfConfigClauseListSyntax
│ ╰─[0]: IfConfigClauseSyntax
│   ├─poundKeyword: poundIfKeyword
│   ├─condition: IdentifierExprSyntax
│   │ ╰─identifier: identifier("MY_FLAG")
│   ╰─elements: CodeBlockItemListSyntax
│     ├─[0]: CodeBlockItemSyntax
│     │ ├─item: MacroExpansionExprSyntax
│     │ │ ├─poundToken: pound
│     │ │ ├─macro: identifier("") MISSING
│     │ │ ╰─argumentList: TupleExprElementListSyntax
│     │ ╰─semicolon: semicolon MISSING
│     ╰─[1]: CodeBlockItemSyntax
│       ╰─item: IdentifierExprSyntax
│         ╰─identifier: identifier("elif")
╰─poundEndif: poundEndifKeyword
```

When inserting the missing identifier, we see that `#` and identifiers don’t need to be separated by a space, so we remove the space. This is really useful for example if you have `someCall(1 2)` where a comma is missing. In this case you want to insert it after immediate after the `1` and not the space.

So now to the next question: Why don’t we add a space after `<#identifier#>`. For this, we look at the fixed up view of the tree and discover that the missing identifier is followed by a semicolon in that view. And identifier and semicolons don’t need to be separated by spaces. So no space that needs to be inserted here. We just suppress the missing semicolon diagnostic because the code item already has an error. Because most of the time the missing semicolon diagnostic is actually really misleading.

I think fix here would be to improve how we recover here in general so that `elif` doesn’t end up in a separate `CodeBlockItem` anymore.